test/bugs.test.js

```diff
@@ -360,4 +364,238 @@ describe('bug fixes', function() {
+      status: 'active'
+    });
+    const citizen = await Citizen.find();
+    const testDoc = citizen.find(x => x.father != null);
+    assert.equal(testDoc.father.fatherType, 'Guest');
+  });
+  it('`refPath` works without autopopulate on the virtual (gh-96)', async function() {
+    const AddressSchema = new Schema(
+      {
+        name: {
+          type: String
+        },
+        status: {
+          type: String,
+          enum: ['active', 'inactive'],
+          default: 'active'
+        }
+      },
+      {
+        timestamps: true,
+        toJSON: { virtuals: true },
+        toObject: { virtuals: true }
+      }
+    );
+
+    AddressSchema.virtual('residentials', {
+      ref: 'Citizen',
+      localField: '_id',
+      foreignField: 'permanentAddress.address',
+      justOne: false,
+      autopopulate: true,
+      match: {
+        status: 'active'
+      },
+      options: {
+        select:
+          'name nId'
+      }
+    });
+
+    AddressSchema.plugin(autopopulate);
+
+
+    const CitizenSchema = new Schema(
+      {
+        nId: {
+          type: String,
+          required: [true, 'Please add national ID card']
+        },
+        name: {
+          type: String,
+          required: [true, 'Please add a name'],
+          trim: true
+        },
+        permanentAddress: {
+          name: {
+            type: String,
+            trim: true
+          },
+          address: {
+            type: mongoose.Schema.ObjectId,
+            ref: 'Address'
+          }
+        },
+        father: {
+          type: mongoose.Schema.ObjectId,
+          refPath: 'fatherType'
+        },
+        fatherType: {
+          type: String,
+          enum: ['Citizen', 'Guest'],
+          required: true
+        },
+        status: {
+          type: String,
+          enum: ['active', 'inactive'],
+          default: 'active'
+        }
+      },
+      {
+        timestamps: true,
+        toJSON: { virtuals: true },
+        toObject: { virtuals: true }
+      }
+    );
+
+    CitizenSchema.plugin(autopopulate);
+
+
+    const Address = db.model('Address', AddressSchema);
+
+    const Citizen = db.model('Citizen', CitizenSchema);
+    await Address.deleteMany({});
+    await Citizen.deleteMany({});
+    const entry = await Address.create({
+      name: 'Another name for The Address',
+      status: 'active'
+    });
+
+    const doc = await Citizen.create({
+      nId: 'Hello',
+      name: 'There',
+      permanentAddress: {
+        name: 'The Address',
+        address: entry._id
+      },
+      fatherType: 'Guest'
+    });
+    await Citizen.create({
+      nId: 'Yo',
+      name: 'Test',
+      permanentAddress: {
+        name: 'The Address',
+        address: entry._id
+      },
+      father: doc._id,
+      fatherType: 'Citizen',
+      status: 'active'
+    });
+    const addr = await Address.findOne().populate({ path: 'residentials', populate: { path: 'permanentAddress', populate: 'address'} });
```

Comment:
Issue 2: why are you populating `permanentAddress` when that isn't a ref? This should be `path: 'permanentAddress.address'`

Reply:
https://mongoosejs.com/docs/populate.html#deep-populate I was trying to follow the docs example for how to populate in that case. I thought that was the syntax to populate the desired field.